pkg/update/update.go

```diff
@@ -74,15 +75,16 @@ type UpdateSts struct {
 // UpdateTimer encapsulates everything timer and polling related we need to update
 // a StatefulSet.
 type UpdateTimer struct {
-	podUpdateTimeout          time.Duration
-	podMaxPollingInterval     time.Duration
-	sleeper                   Sleeper
-	waitUntilAllPodsReadyFunc func(context.Context, *zap.Logger) error
+	podUpdateTimeout      time.Duration
+	podMaxPollingInterval time.Duration
+	sleeper               Sleeper
+	// TODO check that this func is actually correct
```

Comment:
It might be more correct for it to just check `.Spec.Replicas` against `.Status.ReadyReplicas`?

Reply:
Legacy code 🤷‍♂️